### ISSUE TITLE
Plugin: Synchronize PHP changes for WordPress 6.4 release

### ISSUE THREAD

**Issue by mikachan** [WordPress/gutenberg#54177]
The issue tracks changes to PHP files that (might) need to be backported and merged to Core for the WordPress 6.4 release. The list is exhaustive so it may cover multiple PRs and experimental changes that won't be in 6.4.

## Beta 1 is September 26, 2023

- [Release schedule](https://make.wordpress.org/core/6-4/)
- [6.4 Roadmap](https://make.wordpress.org/core/2023/08/22/roadmap-to-6-4/)

## Tips 👀 

- **Please [open a new Core Trac ticket for each PR](https://core.trac.wordpress.org/newticket)** and link it next to the citation in this issue. 🙇 
- For ease of cross-linking and tracking, you can submit your PRs to [WordPress/wordpress-develop](https://github.com/WordPress/wordpress-develop). Refer to docs on [how to fork the repository and create a PR](https://make.wordpress.org/core/handbook/tutorials/trac/submitting-a-patch/#adding-your-github-fork-to-your-wp-trunk-copy). Link any backport PR next to the citation in this issue. 🙇 
- Try to limit PRs to one feature to make reviewing easier and the commit history clear. See [why](https://github.com/WordPress/gutenberg/issues/47187#issuecomment-1384107056)
- Modifications/additions to PHP unit tests aren't listed: any unit tests introduced in the cited PRs should also be ported over where applicable.
- Once your backport PR is up, add it to the [WordPress 6.4 Editor Tasks project board](https://github.com/orgs/WordPress/projects/127) so folks can review it. Thank you!

❗ _This is a living document and will be constantly updated as we approach beta._ Sorry if you were mistakenly pinged! If a file or files have already been backported, please tick them off and link to the backport PR/trac issue.

Or, if the changes are not slated for WordPress 6.4, you can ~~strike~~ them or make a comment in this issue. Thank you! 🙇

----

PRs to be backported will have the **"Needs PHP backport"** label applied. 
<img width="142" alt="Screenshot 2023-06-05 at 10 47 22 am" src="https://github.com/WordPress/gutenberg/assets/6458278/299596a3-c7c3-4894-8fa9-012156985bda">

# 📁 [lib](https://github.com/WordPress/gutenberg/tree/trunk/lib)

### 📄 [blocks.php](https://github.com/WordPress/gutenberg/tree/trunk/lib/blocks.php)
- [x] https://github.com/WordPress/gutenberg/pull/54154 - @Mamaduka | Trac ticket | Core backport PR

### 📄 [class-wp-duotone-gutenberg.php](https://github.com/WordPress/gutenberg/tree/trunk/lib/class-wp-duotone-gutenberg.php)
- [x] ~https://github.com/WordPress/gutenberg/pull/52696 - @anton-vlasenko | Trac ticket | Core backport PR~

### 📄 [class-wp-rest-global-styles-controller-gutenberg.php](https://github.com/WordPress/gutenberg/tree/trunk/lib/class-wp-rest-global-styles-controller-gutenberg.php)
- [x] https://github.com/WordPress/gutenberg/pull/53618 - @ramonjd  | [Trac ticket](https://core.trac.wordpress.org/ticket/59296) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5152)

### 📄 [class-wp-theme-json-gutenberg.php](https://github.com/WordPress/gutenberg/tree/trunk/lib/class-wp-theme-json-gutenberg.php)
- [x] https://github.com/WordPress/gutenberg/pull/53259 - @tellthemachines | [Trac ticket](https://core.trac.wordpress.org/ticket/59295) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5148)
- [x] https://github.com/WordPress/gutenberg/pull/53378 - @tellthemachines | [Trac ticket](https://core.trac.wordpress.org/ticket/59294) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5147)
- [x] https://github.com/WordPress/gutenberg/pull/53466 - @dsas | [Trac ticket](https://core.trac.wordpress.org/ticket/59108) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5013)
- [x] https://github.com/WordPress/gutenberg/pull/53568 - @oandregal  | [Trac ticket](https://core.trac.wordpress.org/ticket/59304) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5162)
- [x] https://github.com/WordPress/gutenberg/pull/53574 - @oandregal  | [Trac ticket](https://core.trac.wordpress.org/ticket/59304) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5162)
- [x] [Patterns: Apply layout and alignment to synced patterns in the editor #54416](https://github.com/WordPress/gutenberg/pull/54416) - @aaronrobertshaw | [Trac ticket](https://core.trac.wordpress.org/ticket/59359) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5222)
- [x] https://github.com/WordPress/gutenberg/pull/54509 - @artemiomorales | [Trac ticket](https://core.trac.wordpress.org/ticket/59393) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5248)
- [x] https://github.com/WordPress/gutenberg/pull/53602 - @t-hamano | [Trac ticket](https://core.trac.wordpress.org/ticket/59499)  (has a patch)

### 📄 [client-assets.php](https://github.com/WordPress/gutenberg/tree/trunk/lib/client-assets.php)
- [x] ~https://github.com/WordPress/gutenberg/pull/50387 - @tyxla  | Trac ticket | Core backport PR (experimental)~
- [x] ~https://github.com/WordPress/gutenberg/pull/52455 - @peterwilsoncc | Trac ticket | Core backport PR~
- [x] https://github.com/WordPress/gutenberg/pull/53161 - @glendaviesnz | [Trac ticket](https://core.trac.wordpress.org/ticket/59380) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5235)

### 📄 [experiments-page.php](https://github.com/WordPress/gutenberg/tree/trunk/lib/experiments-page.php)
- [x] https://github.com/WordPress/gutenberg/pull/51449 - @ockham  | [Trac ticket](https://core.trac.wordpress.org/ticket/59313) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5158) (experimental)
- [x] ~https://github.com/WordPress/gutenberg/pull/52681 - @youknowriad | Trac ticket | Core backport PR (experimental)~
- [x] ~https://github.com/WordPress/gutenberg/pull/53241 - @c4rl0sbr4v0 | Trac ticket | Core backport PR (experimental)~ Should not be ported. GB-only and experimental API.

### 📄 [load.php](https://github.com/WordPress/gutenberg/tree/trunk/lib/load.php)
- [x] ~https://github.com/WordPress/gutenberg/pull/52485 - @hellofromtonya | Trac ticket | Core backport PR~ Plugin only
- [x] ~https://github.com/WordPress/gutenberg/pull/52370 - @c4rl0sbr4v0 | Trac ticket | Core backport PR (experimental)~ Should not be ported. It will be reverted in [Image block: Revise lightbox UI to remove 'behaviors'](https://github.com/WordPress/gutenberg/pull/53851/)
- [x] https://github.com/WordPress/gutenberg/pull/53405 - @ramonjd | [Trac ticket](https://core.trac.wordpress.org/ticket/59041) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/4987) (Is this going into WP 6.3.2?)
- [x] ~https://github.com/WordPress/gutenberg/pull/52704 - @matiasbenedetto @hellofromtonya | [Trac ticket](https://core.trac.wordpress.org/ticket/59166) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5285)~
- [x] ~https://github.com/WordPress/gutenberg/pull/53747 - @hellofromtonya | [Trac ticket](https://core.trac.wordpress.org/ticket/59166) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5285)~
- [x] https://github.com/WordPress/gutenberg/pull/53163 - @glendaviesnz | [Trac ticket ](https://core.trac.wordpress.org/ticket/59379#ticket) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5234)
- [x] ~https://github.com/WordPress/gutenberg/pull/53780 - @matiasbenedetto @hellofromtonya | [Trac ticket](https://core.trac.wordpress.org/ticket/59166) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5285)~
- [x] https://github.com/WordPress/gutenberg/pull/53858 - @matiasbenedetto @hellofromtonya | [Trac ticket](https://core.trac.wordpress.org/ticket/59165) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5051)
- [x] https://github.com/WordPress/gutenberg/pull/53931 - @ramonjd | [Trac ticket](https://core.trac.wordpress.org/ticket/59219) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5101)
- [x] ~https://github.com/WordPress/gutenberg/pull/54098 - @matiasbenedetto @hellofromtonya | [Trac ticket](https://core.trac.wordpress.org/ticket/59166) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5285)~
- [x] ~https://github.com/WordPress/gutenberg/pull/54103 - @hellofromtonya | Trac ticket | Core backport PR~ Plugin only

### 📄 [load.php](https://github.com/WordPress/gutenberg/tree/trunk/lib/load.php)
- [x] ~https://github.com/WordPress/gutenberg/pull/52485 - @hellofromtonya | Trac ticket | Core backport PR~ Plugin only

### 📄 [theme.json](https://github.com/WordPress/gutenberg/tree/trunk/lib/theme.json)
- [x] https://github.com/WordPress/gutenberg/pull/50822 - @carolinan | [Trac ticket](https://core.trac.wordpress.org/ticket/59306) | [Core backport PR](https://core.trac.wordpress.org/ticket/59306)

-----

### 📁 [background.php](https://github.com/WordPress/gutenberg/tree/trunk/lib/block-supports/background.php)

- [x] https://github.com/WordPress/gutenberg/pull/53934 - @andrewserong | [Trac ticket](https://core.trac.wordpress.org/ticket/59357) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5209)
- [x] https://github.com/WordPress/gutenberg/pull/51116 - @aristath | [Trac Ticket](https://core.trac.wordpress.org/ticket/59405) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5244)

### 📄 [behaviors.php](https://github.com/WordPress/gutenberg/tree/trunk/lib/block-supports/behaviors.php)
- [x] https://github.com/WordPress/gutenberg/pull/52067 - @artemiomorales | Trac ticket | Core backport PR (experimental)
- [x] https://github.com/WordPress/gutenberg/pull/52178 - @artemiomorales | Trac ticket | Core backport PR (experimental)
- [x] https://github.com/WordPress/gutenberg/pull/52765 - @artemiomorales | Trac ticket | Core backport PR (experimental)

### 📄 [colors.php](https://github.com/WordPress/gutenberg/tree/trunk/lib/block-supports/colors.php)
- [x] https://github.com/WordPress/gutenberg/pull/53667 - @aaronrobertshaw | [Trac ticket](https://core.trac.wordpress.org/ticket/59309) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5164)

### 📄 [elements.php](https://github.com/WordPress/gutenberg/tree/trunk/lib/block-supports/elements.php)
- [x] https://github.com/WordPress/gutenberg/pull/53667 - @aaronrobertshaw | [Trac ticket](https://core.trac.wordpress.org/ticket/59309) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5164)

### 📄 [layout.php](https://github.com/WordPress/gutenberg/tree/trunk/lib/block-supports/layout.php)
- [x] https://github.com/WordPress/gutenberg/pull/54075 - @dmsnell | [Trac ticket](https://core.trac.wordpress.org/ticket/59443) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5252)

### 📄 [typography.php](https://github.com/WordPress/gutenberg/tree/trunk/lib/block-supports/typography.php)
- [x] https://github.com/WordPress/gutenberg/pull/53081 - @ramonjd | [Trac ticket](https://core.trac.wordpress.org/ticket/59048) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/4989) 

-----

### 📄 [block-patterns.php](https://github.com/WordPress/gutenberg/blob/trunk/lib/compat/wordpress-6.4/block-patterns.php)
- [x] https://github.com/WordPress/gutenberg/pull/53163 - @glendaviesnz | [Trac ticket ](https://core.trac.wordpress.org/ticket/59379#ticket) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5234)

### 📄 [blocks.php](https://github.com/WordPress/gutenberg/tree/trunk/lib/compat/wordpress-6.4/blocks.php)
- [x] https://github.com/WordPress/gutenberg/pull/53405 - @ramonjd | [Trac ticket](https://core.trac.wordpress.org/ticket/59041) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/4987) (Is this going into WP 6.3.2?)

### 📄 [class-gutenberg-rest-global-styles-revisions-controller-6](https://github.com/WordPress/gutenberg/blob/trunk/lib/compat/wordpress-6.4/class-gutenberg-rest-global-styles-revisions-controller-6-4.php)
- [x] ~https://github.com/WordPress/gutenberg/pull/52370 - @c4rl0sbr4v0 | Trac ticket | Core backport PR (experimental)~ Should not be ported. It will be reverted in [Image block: Revise lightbox UI to remove 'behaviors'](https://github.com/WordPress/gutenberg/pull/53851/)

### 📄 [rest-api.php](https://github.com/WordPress/gutenberg/blob/trunk/lib/compat/wordpress-6.4/rest-api.php)
- [x] ~https://github.com/WordPress/gutenberg/pull/52370 - @c4rl0sbr4v0 | Trac ticket | Core backport PR (experimental)~. Should not be ported. It will be reverted in [Image block: Revise lightbox UI to remove 'behaviors'](https://github.com/WordPress/gutenberg/pull/53851/).
- [x] https://github.com/WordPress/gutenberg/pull/53618 - @ramonjd  | [Trac ticket](https://core.trac.wordpress.org/ticket/59296) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5152)

### 📄 [script-loader.php](https://github.com/WordPress/gutenberg/blob/trunk/lib/compat/wordpress-6.4/script-loader.php)
- [x] https://github.com/WordPress/gutenberg/pull/53931 - @ramonjd | [Trac ticket](https://core.trac.wordpress.org/ticket/59219) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5101)
- [x] https://github.com/WordPress/gutenberg/pull/54254 - @ellatrix | [Trac ticket](https://core.trac.wordpress.org/ticket/59456) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5309) | [Core commit](https://core.trac.wordpress.org/changeset/56736)

### 📄 [kses.php](https://github.com/WordPress/gutenberg/blob/trunk/lib/compat/wordpress-6.4/kses.php)
- [x] https://github.com/WordPress/gutenberg/pull/54581 - @t-hamano | [Trac ticket](https://core.trac.wordpress.org/ticket/59387) (has a patch)
-----

### 📄 [auto-inserting-blocks.php](https://github.com/WordPress/gutenberg/blob/trunk/lib/experimental/auto-inserting-blocks.php)
- [x] https://github.com/WordPress/gutenberg/pull/51449 - @ockham  | [Trac ticket](https://core.trac.wordpress.org/ticket/59313) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5158) (experimental)
- [x] https://github.com/WordPress/gutenberg/pull/53183 - @ockham  | [Trac ticket](https://core.trac.wordpress.org/ticket/59313) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5158) (experimental) 
- [x] https://github.com/WordPress/gutenberg/pull/52969 - @ockham | [Trac ticket](https://core.trac.wordpress.org/ticket/59313) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5158) (experimental)

### 📄 [blocks.php](https://github.com/WordPress/gutenberg/blob/trunk/lib/experimental/blocks.php)
- [x] ~https://github.com/WordPress/gutenberg/pull/53247 - @michalczaplinski | Trac ticket | Core backport PR~. Plugin-only and experimental.
- [x] ~https://github.com/WordPress/gutenberg/pull/52579 - @westonruter  | Trac ticket | Core backport PR~ (not relevant as Interactivity API is not part of core merge yet)

### 📄 [class--wp-editors.php](https://github.com/WordPress/gutenberg/blob/trunk/lib/experimental/class--wp-editors.php)
- [x] ~https://github.com/WordPress/gutenberg/pull/52696 - @anton-vlasenko | Trac ticket | Core backport PR~
- [x] ~https://github.com/WordPress/gutenberg/pull/50387 - @tyxla  | Trac ticket | Core backport PR (experimental)~

### 📄 [class-gutenberg-rest-block-patterns-controller.php](https://github.com/WordPress/gutenberg/blob/trunk/lib/experimental/class-gutenberg-rest-block-patterns-controller.php)
- [x] https://github.com/WordPress/gutenberg/pull/51449 - @ockham  | [Trac ticket](https://core.trac.wordpress.org/ticket/59313) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5158) (experimental)

### 📄 [class-gutenberg-rest-template-revision-count.php](https://github.com/WordPress/gutenberg/blob/trunk/lib/experimental/class-gutenberg-rest-template-revision-count.php)
- [x] ~~https://github.com/WordPress/gutenberg/pull/51774~~ (Will probably not make 6.4. Blocked by [upstream](https://core.trac.wordpress.org/ticket/57704#comment:4))

### 📄 [class-wp-rest-block-editor-settings-controller.php](https://github.com/WordPress/gutenberg/blob/trunk/lib/experimental/class-wp-rest-block-editor-settings-controller.php)
- [x] ~https://github.com/WordPress/gutenberg/pull/52696 - @anton-vlasenko | Trac ticket | Core backport PR~
- [x] https://github.com/WordPress/gutenberg/pull/52045 - @spacedmonkey | Trac ticket | Core backport PR

### 📄 [class-wp-rest-customizer-nonces.php](https://github.com/WordPress/gutenberg/blob/trunk/lib/experimental/class-wp-rest-customizer-nonces.php)
- [x] ~https://github.com/WordPress/gutenberg/pull/52696 - @anton-vlasenko | Trac ticket | Core backport PR~

### 📄 [disable-tinymce.php](https://github.com/WordPress/gutenberg/blob/trunk/lib/experimental/disable-tinymce.php)
- [x] ~https://github.com/WordPress/gutenberg/pull/52325 - @tyxla  | Trac ticket | Core backport PR (experimental)~
- [x] ~https://github.com/WordPress/gutenberg/pull/52935 - @tyxla  | Trac ticket | Core backport PR (experimental)~
- [x] ~https://github.com/WordPress/gutenberg/pull/52938 - @tyxla  | Trac ticket | Core backport PR (experimental)~

### 📄 [editor-settings.php](https://github.com/WordPress/gutenberg/blob/trunk/lib/experimental/editor-settings.php)
- [x] ~https://github.com/WordPress/gutenberg/pull/50387 - @tyxla  | Trac ticket | Core backport PR (experimental)~
- [x] ~https://github.com/WordPress/gutenberg/pull/52833 - @c4rl0sbr4v0  | Trac ticket | Core backport PR (experimental)~ (not relevant as Interactivity API is not part of core merge yet)
- [x] ~https://github.com/WordPress/gutenberg/pull/52681 - @youknowriad | Trac ticket | Core backport PR (experimental)~
- [x] ~https://github.com/WordPress/gutenberg/pull/53241 - @c4rl0sbr4v0 | Trac ticket | Core backport PR (experimental)~ Should not be ported. GB-only and experimental API.
- [x] ~https://github.com/WordPress/gutenberg/pull/53527 - @ockham | Trac ticket | Core backport PR (experimental)~
- [x] https://github.com/WordPress/gutenberg/pull/52969 - @ockham | [Trac ticket](https://core.trac.wordpress.org/ticket/59313) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5158) (experimental)

### 📄 [kses.php](https://github.com/WordPress/gutenberg/blob/trunk/lib/experimental/kses.php)
- [x] ~https://github.com/WordPress/gutenberg/pull/52696 - @anton-vlasenko | Trac ticket | Core backport PR~

### 📄 [rest-api.php](https://github.com/WordPress/gutenberg/blob/trunk/lib/experimental/rest-api.php)
- [x] ~https://github.com/WordPress/gutenberg/pull/52696 - @anton-vlasenko | Trac ticket | Core backport PR~
- [x] ~~https://github.com/WordPress/gutenberg/pull/51774~~ (Will probably not make 6.4. Blocked by [upstream](https://core.trac.wordpress.org/ticket/57704#comment:4))

### 📄 [synchronization.php](https://github.com/WordPress/gutenberg/blob/trunk/lib/experimental/synchronization.php)
- [x] ~https://github.com/WordPress/gutenberg/pull/52681 - @youknowriad | Trac ticket | Core backport PR (experimental)~

~### 📁 [lib/experimental/fonts-api](https://github.com/WordPress/gutenberg/tree/trunk/lib/experimental/fonts-api)~
The Fonts API will not be backported to Core, as it's been deprecated in the plugin and replaced by Font Face.

~#### 📄 [fonts-api.php](https://github.com/WordPress/gutenberg/blob/trunk/lib/experimental/fonts-api/fonts-api.php)~
- [x] ~https://github.com/WordPress/gutenberg/pull/52343 - @hellofromtonya | Trac ticket | Core backport PR~

#### 📄 [class-wp-font-library.php](https://github.com/WordPress/gutenberg/blob/trunk/lib/experimental/fonts/font-library/class-wp-font-library.php)
- [x] ~https://github.com/WordPress/gutenberg/pull/54122 - @jffng @matiasbenedetto | [Trac ticket](https://core.trac.wordpress.org/ticket/59166) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5285)~
- [x] ~https://github.com/WordPress/gutenberg/pull/53986 - @madhusudhand | [Trac ticket](https://core.trac.wordpress.org/ticket/59166) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5285)~

#### 📄 [class-wp-font-family.php](https://github.com/WordPress/gutenberg/blob/trunk/lib/experimental/fonts/font-library/class-wp-font-family.php)
- [x] ~https://github.com/WordPress/gutenberg/pull/54490 - @jffng | [Trac ticket](https://core.trac.wordpress.org/ticket/59166) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5285)~
- [x] ~https://github.com/WordPress/gutenberg/pull/54559 - @matiasbenedetto | [Trac ticket](https://core.trac.wordpress.org/ticket/59166) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5285)~
- [x] ~https://github.com/WordPress/gutenberg/pull/54647 - @madhusudhand | [Trac ticket](https://core.trac.wordpress.org/ticket/59166) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5285)~

#### 📄 [class-wp-font-family-utils.php](https://github.com/WordPress/gutenberg/blob/trunk/lib/experimental/fonts/font-library/class-wp-font-family-utils.php)
- [x] ~https://github.com/WordPress/gutenberg/pull/54435 - @matiasbenedetto | [Trac ticket](https://core.trac.wordpress.org/ticket/59166) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5285)~

-----

## 📁 [fonts-library](https://github.com/WordPress/gutenberg/tree/trunk/phpunit/fonts-library)
- [x] ~https://github.com/WordPress/gutenberg/pull/53702 - @hellofromtonya | [Trac ticket](https://core.trac.wordpress.org/ticket/59166) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5285)~
- [x] ~https://github.com/WordPress/gutenberg/pull/54566 - @matiasbenedetto | [Trac ticket](https://core.trac.wordpress.org/ticket/59166) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5285)~
- [x] https://github.com/WordPress/gutenberg/pull/54615 - @hellofromtonya | [Trac ticket](https://core.trac.wordpress.org/ticket/59165) | [Core backport PR ](https://github.com/WordPress/wordpress-develop/pull/5298) | [Core commit](https://core.trac.wordpress.org/changeset/56688)
-----

### 📄 packages/style-engine/class-wp-style-engine.php
- [x] https://github.com/WordPress/gutenberg/pull/53710 - @ramonjd | [Trac ticket](https://core.trac.wordpress.org/ticket/59401) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5255)
- [x] https://github.com/WordPress/gutenberg/pull/53667 - @aaronrobertshaw | [Trac ticket](https://core.trac.wordpress.org/ticket/59309) | [Core backport PR
](https://github.com/WordPress/wordpress-develop/pull/5164)

-----

### 📄 class-wp-block-parser.php
- [x] https://github.com/WordPress/gutenberg/pull/54424 - @spacedmonkey | [Trac ticket](https://core.trac.wordpress.org/ticket/59168) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/5049)

-----

## 📁 [packages/block-library/](https://github.com/WordPress/gutenberg/tree/trunk/packages/block-library/)
- [x] https://github.com/WordPress/gutenberg/pull/54532 - @glendaviesnz  | Trac ticket | Core backport PR

-----

## 🚧 Contingent features

The following are features that may or may not be shipped with 6.4.

**Comment by Mamaduka** [comment]
Thanks for the ping.

I just double-checked the core files, and there's no need to backport #54154. The [autogenerated static blocks list](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/blocks/require-static-blocks.php) looks correct for the core.

**Comment by mikachan** [comment]
Thanks @Mamaduka! I'll check it off the list.

**Comment by tellthemachines** [comment]
I've added Trac tickets/PRs above to my two changes to [class-wp-theme-json-gutenberg.php](https://github.com/WordPress/gutenberg/tree/trunk/lib/class-wp-theme-json-gutenberg.php). I think @oandregal 's off for the next week or so, so I'm happy to do PRs for his changes to the same file too!

**Comment by ramonjd** [comment]
Thanks for getting this together! I've added all the links to backport PRs/trac tickets for my changes.



<details>

<summary>Summary of updates</summary>


### [Try: bundle WP_REST_Global_Styles_Controller class instead of inheriting per WordPress version #53618](https://github.com/WordPress/gutenberg/pull/53618) 
Mostly a plugin-only refactor. A couple of code quality tweaks. Bundled these with #52819 and added backport links.


### [Patterns: add `delete_posts` to the wp_block (patterns) capabilities #53405](https://github.com/WordPress/gutenberg/pull/53405) - [@ramonjd](https://github.com/ramonjd) | [Trac ticket](https://core.trac.wordpress.org/ticket/59041) | [Core backport PR](https://github.com/WordPress/wordpress-develop/pull/4987) (Is this going into WP 6.3.2?)

Not sure, but let's keep it here since everything is ready in case it can go into 6.4.

### [Date: add relative time translations for moment.js #53931](https://github.com/WordPress/gutenberg/pull/53931)

Done. Added links

###  [Fluid typography: add min and max viewport width configurable options #53081](https://github.com/WordPress/gutenberg/pull/53081)

Done. Added links

### [Rest API: rename navigation fallback classes from WP_ to Gutenberg_ #51959](https://github.com/WordPress/gutenberg/pull/51959) 

Plugin only changes - removed

### [Template revisions API: move back to experimental #51774](https://github.com/WordPress/gutenberg/pull/51774)

I don't think this one is going make it. It's blocked:
https://core.trac.wordpress.org/ticket/57704#comment:4

I struck it out and left it there for visibility


</details>

**Comment by tyxla** [comment]
Thanks for the ping! 🙌 

I've taken a thorough look at my PRs and none of them need to be backported. They're all related to an experiment to deprecate TinyMCE which isn't ready for public use in core just yet.

**Comment by mikachan** [comment]
Thanks @tyxla! I've put a strikethrough through your PRs.

**Comment by hellofromtonya** [comment]
The Fonts API is deprecated and replaced by Font Face to work with the Font Library. Thus, the Fonts API found in `lib/experimental/fonts-api/` will not be backported to Core. Plus these files will be removed from Gutenberg once the Font Library is turned on.

Description updated ✅

**Comment by ockham** [comment]
I've started work on porting the PHP code for auto-inserting blocks (which have been recently [renamed](https://github.com/WordPress/gutenberg/pull/54147) to "block hooks"). My (very early) WIP PR is [here](https://github.com/WordPress/wordpress-develop/pull/5158); I might have to break it down into smaller ones eventually.

(My only [other change](https://github.com/WordPress/gutenberg/pull/53527) in the above list doesn't need any porting to Core, since it was a mere cleanup on the Gutenberg side to remove something that had already been ported to Core.

**Comment by youknowriad** [comment]
For the sync package, I don't think there's anything to do at the moment as the usage of this package is behind an experimental flag still.

**Comment by michalczaplinski** [comment]
I've updated all of my items and the ones for @c4rl0sbr4v0, as he's on parental leave.

- https://github.com/WordPress/gutenberg/pull/52370
- https://github.com/WordPress/gutenberg/pull/52833
  - Behaviors are not going into 6.4 and will be removed in [Image block: Revise lightbox UI to remove 'behaviors'](https://github.com/WordPress/gutenberg/pull/53851)

- https://github.com/WordPress/gutenberg/pull/53241
- https://github.com/WordPress/gutenberg/pull/53247
  - Anything related to connecting attributes to custom fields is experimental and plugin-only.

**Comment by artemiomorales** [comment]
Hi! We’re still working on the lightbox, so I’ll open the backports for the following once we merge [this current PR](https://github.com/WordPress/gutenberg/pull/54071#discussion_r1317175305).

- https://github.com/WordPress/gutenberg/pull/52067
- https://github.com/WordPress/gutenberg/pull/52178
- https://github.com/WordPress/gutenberg/pull/52765

**Comment by tellthemachines** [comment]
And here's the PR for the remaining 2 theme json class changes: https://github.com/WordPress/wordpress-develop/pull/5162

**Comment by aaronrobertshaw** [comment]
I've added the [Trac ticket](https://core.trac.wordpress.org/ticket/59309) and [PR](https://github.com/WordPress/wordpress-develop/pull/5164) for the changes relating to:
- https://github.com/WordPress/gutenberg/pull/53667

**Comment by carolinan** [comment]
[Trac ticket ](https://core.trac.wordpress.org/ticket/59306) for https://github.com/WordPress/gutenberg/pull/50822 (writing mode)

**Comment by andrewserong** [comment]
I've opened a [Trac ticket](https://core.trac.wordpress.org/ticket/59357) and [PR](https://github.com/WordPress/wordpress-develop/pull/5209) for the changes from:

* https://github.com/WordPress/gutenberg/pull/53934

**Comment by aaronrobertshaw** [comment]
A [Trac ticket](https://core.trac.wordpress.org/ticket/59359) and [PR](https://github.com/WordPress/wordpress-develop/pull/5222) are available for the  changes from:
- https://github.com/WordPress/gutenberg/pull/54416

**Comment by tellthemachines** [comment]
I've opened a [trac ticket](https://core.trac.wordpress.org/ticket/59358) (plus linked PR) for [this bug](https://github.com/WordPress/gutenberg/pull/54485) I just fixed today 😄

**Comment by t-hamano** [comment]
A [Trac ticket](https://core.trac.wordpress.org/ticket/59387) and a patch are available for the changes from:

- https://github.com/WordPress/gutenberg/pull/54581

**Comment by michalczaplinski** [comment]
- https://github.com/WordPress/gutenberg/pull/54509

I've created a [Trac ticket](https://core.trac.wordpress.org/ticket/59393) and [PR](https://github.com/WordPress/wordpress-develop/pull/5248).

**Comment by spacedmonkey** [comment]
This change will also need to backported - https://github.com/WordPress/gutenberg/pull/51116

**Comment by Mamaduka** [comment]
@spacedmonkey, I think Ari is already working on the core patch - https://github.com/WordPress/wordpress-develop/pull/5244.

**Comment by aristath** [comment]
> @spacedmonkey, I think Ari is already working on the core patch - [WordPress/wordpress-develop#5244](https://github.com/WordPress/wordpress-develop/pull/5244).

Yep, trac ticket created in https://core.trac.wordpress.org/ticket/59405 👍

**Comment by michalczaplinski** [comment]
As Behaviors have been removed from the plugin in https://github.com/WordPress/gutenberg/pull/53851 and will not be shipping in 6.4, I marked the following as done:

> - [x] https://github.com/WordPress/gutenberg/pull/52067 - @artemiomorales | Trac ticket | Core backport PR (experimental)
> - [x] https://github.com/WordPress/gutenberg/pull/52178 - @artemiomorales | Trac ticket | Core backport PR (experimental)
> - [x] https://github.com/WordPress/gutenberg/pull/52765 - @artemiomorales | Trac ticket | Core backport PR (experimental)

**Comment by anton-vlasenko** [comment]
The pull request https://github.com/WordPress/gutenberg/pull/52696 doesn't need to be backported to Core. It's specific to Gutenberg.

**Comment by mikachan** [comment]
@peterwilsoncc, could you please confirm if https://github.com/WordPress/gutenberg/pull/52455 needs a backport ahead of Beta 1?

@spacedmonkey, could you please confirm if https://github.com/WordPress/gutenberg/pull/52045 needs a backport ahead of Beta 1?

**Comment by peterwilsoncc** [comment]
> @peterwilsoncc, could you please confirm if #52455 needs a backport ahead of Beta 1?

@mikachan It does not, the NUX script and styles are already included in WP Core. I'll edit this ticket accordingly.

**Comment by mikachan** [comment]
@getdave, I noticed that you approved https://github.com/WordPress/gutenberg/pull/52045 - do you know if it needs a backport ahead of Beta 1?

**Comment by SiobhyB** [comment]
### Status Update

I went through the remaining PRs labelled `Needs PHP backport` and closed those that had been backported. At the time of writing, the following ~19~ 18 issues with the label are still pending. 

#### Monitoring (on this issue's list)

* https://github.com/WordPress/gutenberg/pull/51116 - @aristath | [Trac Ticket](https://core.trac.wordpress.org/ticket/59405) | https://github.com/WordPress/wordpress-develop/pull/5244
* https://github.com/WordPress/gutenberg/pull/53161 - @glendaviesnz | [Trac ticket](https://core.trac.wordpress.org/ticket/59380) | https://github.com/WordPress/wordpress-develop/pull/5235
* ~https://github.com/WordPress/gutenberg/pull/52045 - @spacedmonkey | @getdave asked for status update [here](https://github.com/WordPress/gutenberg/pull/52045#issuecomment-1733202795)~
* https://github.com/WordPress/gutenberg/pull/53747 - @hellofromtonya 
* https://github.com/WordPress/gutenberg/pull/52704 - @matiasbenedetto @hellofromtonya
* https://github.com/WordPress/gutenberg/pull/53780 - @matiasbenedetto @hellofromtonya 
* https://github.com/WordPress/gutenberg/pull/54098 - @matiasbenedetto @hellofromtonya 
* https://github.com/WordPress/gutenberg/pull/54122 - @jffng @matiasbenedetto

#### New (not yet on this issue's list)

I'll go ahead to add the following to the main list on this issue, but also wanted to quickly list them here as a heads up for the authors.

* https://github.com/WordPress/gutenberg/pull/53702 - @hellofromtonya 
* https://github.com/WordPress/gutenberg/pull/53986 - @madhusudhand 
* https://github.com/WordPress/gutenberg/pull/54254 - @ellatrix 
* https://github.com/WordPress/gutenberg/pull/54424 - @spacedmonkey 
* https://github.com/WordPress/gutenberg/pull/54435 - @matiasbenedetto 
* https://github.com/WordPress/gutenberg/pull/54490 - @jffng 
* https://github.com/WordPress/gutenberg/pull/54532 - @glendaviesnz 
* https://github.com/WordPress/gutenberg/pull/54559 - @matiasbenedetto 
* https://github.com/WordPress/gutenberg/pull/54566 - @matiasbenedetto 
* https://github.com/WordPress/gutenberg/pull/54615 - @hellofromtonya 
* https://github.com/WordPress/gutenberg/pull/54647 - @madhusudhand

**Comment by spacedmonkey** [comment]
[Add caching to schema of REST API. #52045](https://github.com/WordPress/gutenberg/pull/52045) was committed in WordPress 6.3 - See https://github.com/WordPress/wordpress-develop/commit/7faab129f56779d0576bed37d025ef16bcb95bf7

**Comment by spacedmonkey** [comment]
Block API: Unnecessary JSON decoding in block parser #54424 - Will be committed to core once the npm packages are updated in core.

**Comment by Mamaduka** [comment]
[Ticket](https://core.trac.wordpress.org/ticket/59456#ticket) and [sync PR](https://github.com/WordPress/wordpress-develop/pull/5309) for - https://github.com/WordPress/gutenberg/pull/54254

**Comment by spacedmonkey** [comment]
Another backport issue - https://github.com/WordPress/gutenberg/pull/54835 - Core ticket https://core.trac.wordpress.org/ticket/59453#ticket

**Comment by t-hamano** [comment]
A [Trac ticket](https://core.trac.wordpress.org/ticket/59499) and a patch are available for the changes from:

- https://github.com/WordPress/gutenberg/pull/53602

**Comment by dmsnell** [comment]
Added the links to the backport of Tag Processor usage in block supports layout; was merged into Core already in WordPress/wordpress-develop#5252

**Comment by mikachan** [comment]
WP 6.4 has been released 🥳 So I'm going to close this. Thanks for everyone's input with the backports!
